noggin/tests/unit/controller/test_user_otp.py

```diff
@@ -157,10 +158,10 @@ def test_user_settings_otp_check_description_escaping(
     page = BeautifulSoup(result.data, "html.parser")
     otp_uri = page.select_one("input#otp-uri")
 
+    # the escaping happens before the secret, so just check that
     assert (
-        otp_uri['value'] == "otpauth://totp/dummy@example.com:pants%20token?issuer="
-        "dummy%40EXAMPLE.COM&secret=L4PD6EXABBJDSCAKS6MZQWT4RSP3PM3QW6H57UHIKFCN7I3"
-        "FGKSHZCCO&digits=6&algorithm=SHA512&period=30"
+        otp_uri['value'][:81]
+        == "otpauth://totp/dummy@example.com:pants%20token?issuer=dummy%40EXAMPLE.COM&secret="
```

Comment:
hardcoding the secret here was causing issues. b ut because we are just testing for the escaping of the token name and the issuer, I truncated the string before the secret hash so it should work across cassette rebuilds.

Reply:
As I understand it, we're mainly interested in that the `description` part in the URI is escaped.

Instead of dealing with the vagaries of how the URI is constructed (I could imagine that order of query params isn't terribly stable), how about we use `urllib.parse.urlparse()` and only check the parts we care about? E.g.:

````
from urllib.parse import parse_qs, urlparse
...
    otp_uri = page.select_one("input#otp-uri")
    parsed_otp_uri = urlparse(otp_uri)
    parsed_query = parse_qs(parsed_otp_uri.query)

    # Not sure we need all of these checked
    assert parsed_otp_uri.scheme == "otpauth"
    assert parsed_otp_uri.netloc == "totp"
    assert parsed_otp_uri.path == "/dummy@example.com:pants%20token"

    assert parsed_query["issuer"] == "dummy@EXAMPLE.COM"
````